### ISSUE TITLE
github Copilot support

### ISSUE THREAD

**Issue by bbhxwl** [gitpod-io/gitpod#10032]
github Copilot support

**Comment by pawlean** [comment]
Hi @bbhxwl! Copilot is GitHub only, I recommend you check out [Tabnine](https://www.tabnine.com/).

**Comment by pawlean** [comment]
Closing this for now, but feel free to open it if you have any questions! Also, please drop by our community [here](https://www.gitpod.io/chat) We'd love to chat some more there!

**Comment by axonasif** [comment]
> github Copilot support

You can open your workspace in Desktop VSCODE to use it.
See https://www.gitpod.io/docs/ides-and-editors/vscode

**Comment by SethG911** [comment]
> Hi @bbhxwl! Copilot is GitHub only, I recommend you check out [Tabnine](https://www.tabnine.com/).

This is not true. Github Copilot can be used in a myriad of IDEs and on any repo type, not just Github. I use it in combination with my Gitlab repos all the time and I even use it in other Visual Studio Code Server instances/providers. There is no reason that GitPod should not be able to handle this. I can install the extension manually, but it fails to authenticate. Again, I don't have this issue with any other online IDE provider.

![Screenshot 2022-10-11 102244](https://user-images.githubusercontent.com/1345439/195117679-bada828f-172c-423b-b067-166f0f47da6a.png)

**Comment by axonasif** [comment]
@SethG911 there are licensing issues with using Copilot outside of Github/Microsoft owned interfaces.

You can use it by opening your workspace on local vscode (the Proprietary VSCode, not https://vscodium.com/)

If GitHub ever decides to publish `GitHub Copilot` on https://open-vsx.org, we can officially use it on Gitpod and other VSCode clients.

**Comment by SethG911** [comment]
@axonasif that makes sense, but is there a reason a service like Coder and their self-hosted https://github.com/coder/code-server can run it and search the non-open source extension library, and Gitpod's version of VSCode can't? Is this just a matter of the flavor of VSCode Server each service is running and what licenses are included? Why doesn't Gitpod use one of these other VSCode server type that is compatible with all the licenses? Is it a licensing cost issue, or just wanting to remain "open source"?

Sorry if these are dumb question, I am not super familiar with all the different types of VSCode licensing.

**Comment by axonasif** [comment]
@SethG911 I could not find a source where Coder themselves states that, could you please point me to it?

All I found was this:
- https://coder.com/docs/coder/latest/workspaces/vs-code-extensions#using-the-public-extensions-marketplaces
